### PR TITLE
Clean worker from pool after it was destroyed

### PR DESCRIPTION
**Pull request by Shinigami92**
## PR Checklist

- [x] Please add a description of your changes.
- [x] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [x] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No
- [x] Maybe

If yes please indicate it and migration info into the CHANGELOG.md file.

---

#133

### CODE REVIEW

**Comment by Shinigami92** [comment]
@pioardi @jerome-benoit I managed to fix the `FixedThreadPool`, but sadly the `FixedClusterPool` doesn't work :slightly_frowning_face:  

https://github.com/pioardi/poolifier/pull/146/checks?check_run_id=1894533439#step:7:94

Only 5 of 10 workers gets removed and I don't know why :thinking: 

Could you help me here?

My assumption is that is has something todo with the non-`async` of `destroyWorker` in a `FixedClusterPool` https://github.com/pioardi/poolifier/pull/146/files#diff-e61ef758d7f4cb04f61b87da930386d5fb58ab9979d4317581f15d9145d043e4R61

But a thread worker has a awaitable termination https://github.com/pioardi/poolifier/pull/146/files#diff-4d40c7c91ed316965541d8bcf71d3edc15b60b4253914428ec9a59eb8d614500R50

**Comment by Shinigami92** [comment]
@pioardi @jerome-benoit nevermind, I have fixed it ^^'
I accidentally passed on `'error'` instead of on `'exit'` for cluster pool destroy worker strategy

File: src/pools/cluster/fixed.ts [hunk 1]
```diff
@@ -59,8 +59,8 @@ export class FixedClusterPool<
   }
 
   protected destroyWorker (worker: Worker): void {
+    worker.once('exit', () => this.removeWorker(worker))
```
Comment (jerome-benoit) [inline src/pools/cluster/fixed.ts:62]: I think that callback should be added by default on both worker type. 
The issue might be that we do have a reference to the worker to remove ... 
That way, each time a worker exit for whatever reason, the circular array in the pool is freed instantly.

Comment (Shinigami92) [reply src/pools/cluster/fixed.ts:62]: Oh yay, it's working :slightly_smiling_face:

**Review by pioardi — COMMENTED** [review]
It is ok for me, but would wait for jerome review  :)

**Comment by Shinigami92** [comment]
> It is ok for me, but would wait for jerome review :)

@jerome-benoit provided
- `once(event: 'exit', handler: ExitHandler<this>): void` https://github.com/pioardi/poolifier/pull/146/files#diff-b9e46115d5f689a0a2eaddebefaff0f2eeb7660835af46e32c3b331948a7e28aR27
- `worker.once('exit', () => this.removeWorker(worker))` https://github.com/pioardi/poolifier/pull/146/files#diff-b9e46115d5f689a0a2eaddebefaff0f2eeb7660835af46e32c3b331948a7e28aR291

So he even saw the PR ^^, maybe he just didn't press the approve button

**Review by pioardi — APPROVED** [review]
Approved since that also Jerome alreadu provider feedbacks

**Pull request merged**